reactioncommerce/example-storefront
# feat: auto-create Hydra client on startup

Impact: **minor**
Type: **feature**

## Changes
For local development, there is a script that runs before the application, which creates a Hydra client for the storefront if one does not yet exist. However, this script isn't included in the production image and doesn't run in a production environment, leading to an additional task that a system admin has to do for every new environment.

This PR moves the auto-creation of the Hydra client to be part of the server startup code, such that it runs in all environments always.

## Breaking changes
None

## Testing
1. In `reaction-hydra`, `docker-compose down -v` and then `docker-compose up -d`. This will ensure the Hydra database is empty and has no clients.
2. Start/restart the example storefront and verify you see logging about the Hydra client being created.
3. Restart the example storefront and verify you see logging about the Hydra client already existing.

Perform above steps both with and without the `docker-compose.dev` override.
